api/python/provisioner/commands/validator/validator.py

```diff
@@ -255,3 +295,210 @@ class YumRepoDataValidator(DirValidator):
     def __init__(self):
         super().__init__({Path("repomd.xml"): FileValidator(required=True)},
                          required=True)
+
+
+@attr.s(auto_attribs=True)
+class FileContentScheme(ABC):
+
+    """
+    Abstract file scheme class.
+
+    It can be used to define an abstract interface if necessary.
+    Also it is useful to have the parent class for all file schemes inherited
+    from this one for comparison using `isinstance`, `type` and for `attr`
+    module.
+
+    Attributes
+    ----------
+    _unexpected_attributes: dict
+        Stores all attribute values that were not listed
+        in the class definition.
+
+        Note: This attribute will be useful in the implementation of logic
+        if we need to validate that some attributes should not be listed
+        in the data scheme.
+    """
+
+    def set_unexpected_attributes(self, args: dict):
+        self._unexpected_attrs = args
+
+    def get_unexpected_attributes(self):
+        return getattr(self, "_unexpected_attrs", None)
+
+    _unexpected_attributes = property(get_unexpected_attributes,
+                                      set_unexpected_attributes)
+
+    @classmethod
+    def from_args(cls, data: Union[list, dict]):
+        unexpected_attrs = dict()
+        if isinstance(data, dict):
+            # TODO: it is good to make copy of input data parameter
+            for _attr in (data.keys() - set(a for a in attr.fields_dict(cls))):
+                # NOTE: Remove unexpected attributes from initialization `data`
+                #  dictionary
+                unexpected_attrs[_attr] = data.pop(_attr)
+
+            # If some attributes are missed in `data`, the `attr` module is
+            #  responsible for that validation
+            obj = cls(**data)
+            obj._unexpected_attributes = unexpected_attrs
+            return obj
+        elif isinstance(data, list):
+            return cls(*data)
```

Comment:
don't we expected `unexpected` ones here?

Reply:
In case of positional arguments we don't expected unexptected arguments. We can evaluate length of `data` and compare it with length of `attrs.field`. Do we need it here? This moment, I added initialization by empty unexpected args
```python
elif isinstance(data, list):
    obj = cls(*data)
    obj._unexpected_attributes = unexpected_attrs
    return obj
```